src/org/labkey/test/components/glassLibrary/grids/QueryGrid.java

```diff
@@ -95,11 +158,58 @@ public GridTabBar getGridTabBar()
         return _gridTabBar.orElseThrow();
     }
 
+    // record count
+
     public int getRecordCount()
     {
         return _gridBar.getRecordCount();
     }
 
+    public QueryGrid waitForRecordCount(int expectedCount)
+    {
+        WebDriverWrapper.waitFor(()-> getRecordCount() == expectedCount,
+                "did not get to the expected record count ["+expectedCount+"] in time",  WAIT_FOR_JAVASCRIPT);
```

Comment:
```suggestion
                "Did not get to the expected record count ["+expectedCount+"] in time.",  WAIT_FOR_JAVASCRIPT);
```
I think it would be good to add an interface where the user is allowed to set the timeout limit, especially since the name of the interface is **wait**ForRecordCount.